### PR TITLE
Vulnerability Detector integration tests for Amazon Linux  2022

### LINKED ISSUE #2813: Integration tests for Vulnerability Detector: Amazon Linux 2022 support
### Description

This issue is part of wazuh/wazuh#13034

We will include support for the Amazon Linux 2022 OS, including a new feed – ALAS 2022.

The following tests should cover the new supported OS version:

- [Feed tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_feeds): These tests have to cover the addition of the new feed for `Amazon Linux 2022`
- [Provider tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_providers): Here we will test the settings for `Amazon Linux` with OS version `2022`
- [Scan results tests](https://github.com/wazuh/wazuh-qa/tree/master/tests/integration/test_vulnerability_detector/test_scan_results): Here we have to add the tests that ensure `Amazon Linux 2022` agents are scanned properly

### PR DESCRIPTION
**Pull request by CamiRomero**
|Related issue|
|---|
|Close #2813  |

## Description

For Wazuh v4.4.0 we want to add Amazon Linux 2022 OS support for the Wazuh Vulnerability Detector module.

The following test was added to cover the new supported OS version:

- `Feed tests:` These tests have to cover the addition of the new feed for `Amazon Linux 2022`
- `Provider tests`: Here we test the settings for `Amazon Linux` with OS version `2022`
- `Scan results tests`: Here we  add the tests that ensure `Amazon Linux 2022` agents are scanned properly

Modules Involved: 

- test_vulnerability_detector/test_feeds/test_download_feeds.py
- test_vulnerability_detector/test_feeds/test_duplicate_feeds.py
- test_vulnerability_detector/test_feeds/test_import_invalid_feed_type.py
- test_vulnerability_detector/test_providers/test_enabled.py
- test_vulnerability_detector/test_providers/test_missing_os.py
- test_vulnerability_detector/test_providers/test_os.py
- test_vulnerability_detector/test_providers/test_update_interval.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_and_nvd_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_provider_vulnerabilities.py
- test_vulnerability_detector/test_scan_results/test_scan_vulnerability_removal.py

## Tests Results
| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Jenkins | [:green_circle: ](https://ci.wazuh.info/job/Test_integration/27726/)  [:green_circle: ](https://ci.wazuh.info/job/Test_integration/27739/) [:green_circle: ](https://ci.wazuh.info/job/Test_integration/27740/) | 2022/06/09 | @Deblintrake09  | 
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Local | [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8764512/R3.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/8764510/R1.zip)  [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/8764511/R2.zip)  | 2022/05/24 | @Deblintrake09  | 

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.

- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ Wazuh commit: TBD \
 Release report: TBD
 
 ### Added
-
+- Add VDT tests support for ALAS 2022 provider ([#2649](https://github.com/wazuh/wazuh-qa/pull/2903))
```
Comment (damarisg) [inline CHANGELOG.md:22]:
```suggestion
- Add test_providers cases for Amazon Linux 2022. ([#2649](https://github.com/wazuh/wazuh-qa/pull/2903))
```

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 2]
```diff
@@ -8,6 +8,27 @@
...
+  configuration_parameters:
+    PROVIDER: 'alas'
+    OS: amazon-linux-2
+    UPDATE_INTERVAL: '5s'
+  metadata:
+    provider_name: 'Amazon Linux 2'
+    update_interval: '5s'
+
+- name: 'Amazon_Linux 2022'
+  description: 'Test update interval 5s Amazon Linux'
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml:22]:
```suggestion
  description: 'Test update interval 5s Amazon Linux 2022'
```

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 3]
```diff
@@ -8,6 +8,27 @@
     provider_name: 'Amazon Linux 1'
     update_interval: '5s'
 
+- name: 'Amazon_Linux 2'
+  description: 'Test update interval 5s Amazon Linux'
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml:12]:
```suggestion
  description: 'Test update interval 5s Amazon Linux 2'
```

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-- name: 'Amazon_Linux'
+- name: 'Amazon_Linux 1'
   description: 'Test update interval 5s Amazon Linux'
```
Comment (damarisg) [inline tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml:2]:
```suggestion
  description: 'Test update interval 5s Amazon Linux 1'
```

**Review by damarisg — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml (hunk 4 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: fixed in commit https://github.com/wazuh/wazuh-qa/pull/2911/commits/1b19247695cb3c80a4388c6ca67963e80f244b46

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml (hunk 3 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: Fixed in commit https://github.com/wazuh/wazuh-qa/pull/2911/commits/1b19247695cb3c80a4388c6ca67963e80f244b46

Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: fixed in commit https://github.com/wazuh/wazuh-qa/pull/2911/commits/1b19247695cb3c80a4388c6ca67963e80f244b46

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml (hunk 2 above)
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: Fixed in commit https://github.com/wazuh/wazuh-qa/pull/2911/commits/1b19247695cb3c80a4388c6ca67963e80f244b46

File: CHANGELOG.md (hunk 1 above)
Comment (Deblintrake09) [reply CHANGELOG.md]: Fixed in commit https://github.com/wazuh/wazuh-qa/pull/2911/commits/0b47a85d7ee857afef2bd84c5eaa4fb93ddf9f4a

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_vulnerabilities.yaml [hunk 5]
```diff
@@ -41,11 +41,11 @@
   configuration_parameters: null
   metadata:
     provider_name: 'alas'
-    system: 'ALAS2'
-    json_feed: 'custom_alas2_feed.json'
+    system: 'ALAS2022'
+    json_feed: 'custom_alas2022_feed.json'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_vulnerabilities.yaml:45]: Fixed in commit: https://github.com/wazuh/wazuh-qa/pull/2911/commits/83ad2c6b4854298e219fc5d661d9076226d97973

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_vulnerabilities.yaml [hunk 6]
```diff
@@ -41,11 +41,11 @@
...
     oval_feed: null
     nvd_feed: 'custom_nvd_alternative_feed.json'
-    json_feed_tag: CUSTOM_ALAS2_JSON_FEED
+    json_feed_tag: CUSTOM_ALAS2022_JSON_FEED
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_vulnerabilities.yaml]: Fixed in commit: https://github.com/wazuh/wazuh-qa/pull/2911/commits/83ad2c6b4854298e219fc5d661d9076226d97973

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_and_nvd_vulnerabilities.yaml [hunk 7]
```diff
@@ -41,11 +41,11 @@
...
   metadata:
     provider_name: 'alas'
-    system: 'ALAS2'
-    json_feed: 'custom_alas2_feed.json'
+    system: 'ALAS2022'
+    json_feed: 'custom_alas2022_feed.json'
     oval_feed: null
     nvd_feed: 'custom_nvd_feed.json'
-    json_feed_tag: CUSTOM_ALAS2_JSON_FEED
+    json_feed_tag: CUSTOM_ALAS2022_JSON_FEED
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_provider_and_nvd_vulnerabilities.yaml]: Fixed in commit: https://github.com/wazuh/wazuh-qa/pull/2911/commits/83ad2c6b4854298e219fc5d661d9076226d97973

File: CHANGELOG.md [hunk 8]
```diff
@@ -47,6 +47,8 @@ Release report: TBD
 - Analysisd - add a new test to check analysisd socket properties ([#2405](https://github.com/wazuh/wazuh-qa/pull/2405))
 - Add system test to check synchronization between agent and manager when one of this was stopped. ([#2536](https://github.com/wazuh/wazuh-qa/pull/2536))
 - Add a test to check the multigroups shared file content. ([#2746](https://github.com/wazuh/wazuh-qa/pull/2746))
+- Add test_scan_results cases for Amazon Linux 2022. ([#2905](https://github.com/wazuh/wazuh-qa/pull/2905))
+- Add test_feeds cases for Amazon Linux 2022. ([#2868](https://github.com/wazuh/wazuh-qa/pull/2868))
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply CHANGELOG.md]: https://github.com/wazuh/wazuh-qa/pull/2911/commits/a7ebc5b563ee89b4aa1ae7829d94eaf6888fac43

File: deps/wazuh_testing/wazuh_testing/mocking/__init__.py [hunk 9]
```diff
@@ -68,6 +68,9 @@
               'os_arch': 'x86_64', 'config_sum': '', 'merged_sum': '', 'manager_host': 'alas2', 'node_name': 'node01',
               'date_add': '1645538646', 'last_keepalive': '253402300799', 'sync_status': 'synced',
               'connection_status': 'active'},
+    'ALAS2022': {'hostname': 'alas2022', 'architecture': 'x86_64', 'os_name': 'Amazon Linux', 'os_version': '2022',
+              'os_codename': '', 'os_major': '2022', 'os_minor': '', 'os_patch': '', 'os_build': '', 'os_platform': 'amzn',
+              'sysname': 'Linux', 'release': '5.15.29-16.111.amzn2022.x86_64'},
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/mocking/__init__.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2911/commits/ada362eeb7871640dd3de3c2d5d4317d58f25bc1

File: deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/__init__.py [hunk 10]
```diff
@@ -31,6 +31,7 @@
 CUSTOM_ARCHLINUX_JSON_FEED = 'custom_archlinux_feed.json'
 CUSTOM_ALAS_JSON_FEED = 'custom_alas_feed.json'
 CUSTOM_ALAS2_JSON_FEED = 'custom_alas2_feed.json'
+CUSTOM_ALAS2022_JSON_FEED = 'custom_alas2022_feed.json'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply deps/wazuh_testing/wazuh_testing/modules/vulnerability_detector/__init__.py]: Fixed in https://github.com/wazuh/wazuh-qa/pull/2911/commits/086527a2f97a03e46ed9112e1d74ea5fcc94ef39

File: tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml [hunk 11]
```diff
@@ -1,5 +1,5 @@
-- name: 'Amazon_Linux'
-  description: 'Test update interval 5s Amazon Linux'
+- name: 'Amazon_Linux 1'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_providers/data/test_cases/cases_update_interval.yaml]: Fixed in commit: https://github.com/wazuh/wazuh-qa/commit/83ad2c6b4854298e219fc5d661d9076226d97973

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerability_removal.py [hunk 12]
```diff
@@ -63,35 +65,27 @@
...
 
 @pytest.mark.tier(level=1)
-@pytest.mark.parametrize('configuration, metadata', zip(configurations, configuration_metadata), ids=test_case_ids)
-def test_vulnerability_removal_update_package(configuration, metadata, set_wazuh_configuration_vdt,
+@pytest.mark.parametrize('configuration, metadata, agent_system', zip(configurations, configuration_metadata, systems),
+                         ids=test_case_ids)
+def test_vulnerability_removal_update_package(configuration, metadata, agent_system, set_wazuh_configuration_vdt,
                                               truncate_monitored_files, clean_cve_tables_func,
-                                              prepare_full_scan_with_vuln_packages, setup_log_monitor,
+                                              prepare_full_scan_with_vuln_packages_and_custom_system, setup_log_monitor,
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerability_removal.py:87]: fixed in https://github.com/wazuh/wazuh-qa/pull/2911/commits/a036de102635d4a21adde411a83ac3147922be36

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_vulnerability_removal.yaml [hunk 13]
```diff
@@ -1,16 +1,34 @@
...
+    provider_name: 'alas'
+    system: 'ALAS2022'
+    json_feed: 'custom_alas2022_feed.json'  
+    oval_feed: null
+    nvd_feed: 'custom_nvd_feed.json'
+    json_feed_tag: CUSTOM_ALAS2022_JSON_FEED
+    nvd_feed_tag: CUSTOM_NVD_JSON_FEED
     test_package_vendor: 'WazuhIntegrationTests'
     test_package_version: '1.0.0'
     test_package_version_not_vulnerable: '2.1.0'
```
[the remark this answers is not in the document]
Comment (Deblintrake09) [reply tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_vulnerability_removal.yaml]: Removed **test_package_vendor**
Fixed in https://github.com/wazuh/wazuh-qa/pull/2911/commits/345b6ed632d1923c4aace23b3291b5a7310c4953

**Review by damarisg — APPROVED** [review]
## Tests Results

| Test Path | Os/Type | Execution Type | Results | Date | By |
|--|--|--|--|--|--|
|  test/integration/test_vulnerability_detector/| CentOS - Manager | Jenkins | [:green_circle:](https://ci.wazuh.info/job/Test_integration/26903/) | 2022/05/26 | Seyla  |

**Comment by jmv74211** [comment]
This PR is closed since these related changes have been merges in this PR https://github.com/wazuh/wazuh-qa/pull/2955

**Pull request closed without merging**